CMakeLists.txt

```diff
@@ -143,19 +144,24 @@ find_package(clang 15)
 find_package(lld 15)
 
 if(ZIG_STATIC_ZLIB)
-    list(REMOVE_ITEM LLVM_LIBRARIES "-lz")
+    if (MSVC)
+        list(REMOVE_ITEM LLVM_SYSTEM_LIBRARIES "z.lib")
+    else()
+        list(REMOVE_ITEM LLVM_SYSTEM_LIBRARIES "-lz")
```

Comment:
What is the difference between `LLVM_LIBRARIES` and `LLVM_SYSTEM_LIBRARIES`?  Will this change not potentially break other targets linking against system version of LLVM such as macOS?

Reply:
It is possible to link statically against a system provided LLVM e.g., Homebrew provided LLVM on macOS, in which case, I am wondering if this change doesn't break it as now we are removing `-lz` from `LLVM_SYSTEM_LIBRARIES` rather than `LLVM_LIBRARIES`, assuming they are not the same.

Reply:
I added `LLVM_SYSTEM_LIBRARIES` to hold the result of `llvm-config.exe --system-libs`, instead of adding it to `LLVM_LIBRARIES`. The reason for breaking it out into it's own variable was to handle this difference:

```
e:\dev\zig-bootstrap\out-win\host\bin>llvm-config.exe --system-libs
psapi.lib shell32.lib ole32.lib uuid.lib advapi32.lib z.lib
```

```
$ ./llvm-config.exe  --system-libs
-lpsapi -lshell32 -lole32 -luuid -ladvapi32 -lz
```

The problem was that `addCMakeLibraryList` requires a `-l` prefix in order to pass the library to `exe.linkSystemLibrary` instead of `exe.addObjectFile`. There is no `-l` prefix in the msvc `--system-libs` case. What was happening was ie. `uuid.lib` would be passed to `exe.addObjectFile` which would generate a path like `c:\path\to\zig\uuid.lib` when linking, which would fail.

For comparison the `--libfiles` result on msvc vs msys2/mingw:

```
e:\dev\zig-bootstrap\out-win\host\bin>llvm-config.exe --libfiles
e:\dev\zig-bootstrap\out-win\host\lib\LLVMWindowsManifest.lib e:\dev\zig-bootstrap\out-win\host\lib\LLVMWindowsDriver.lib ... snip
```

```
$ ./llvm-config.exe  --libfiles
E:/dev/zig-bootstrap/out/build-llvm-host/lib/libLLVMWindowsManifest.a E:/dev/zig-bootstrap/out/build-llvm-host/lib/libLLVMWindowsDriver.a ... snip
```
In this case, you do want to pass the full lib path, so I couldn't just check for a .lib extension to make the distinction.

Reply:
To answer your second question, I believe `-lz` should be in `LLVM_SYSTEM_LIBARIES` in the `ZIG_STATIC_LLVM` case on macOS, but I don't have access to a macOS system to check if `-lz` should up in the `llvm-config --libfiles` result or the `llvm-config --system-libs` result.